### ISSUE TITLE
epsilon decay ends too early relative to total training steps

### ISSUE THREAD

**Issue by ArgonZapan** [ArgonZapan/DQN-transformer#11]
## Training Hyperparameter Issue

In `config.toml`, the epsilon decay configuration causes exploration to end well before training completes:

```toml
epsilon_decay_fraction = 0.3
buffer_capacity = 500000
```

Combined with `min_buffer_size = 10000`:
- Training begins meaningful updates after ~10,000 steps
- Epsilon decay completes at step `0.3 * 500000 = 150,000`
- Total training steps: `buffer_capacity = 500,000` (or more with episodes continuing after buffer fills)

This means **for the majority of training (150,000 → 500,000+ steps), epsilon stays frozen at `epsilon_end = 0.05`**.

At 0.05 epsilon, the agent still takes random actions 5% of the time — which is a reasonable exploitation/exploration balance for late training, but it's static. The decay curve can't adapt if the agent learns faster or slower than expected.

## Expected Behavior

Epsilon should continue decaying (even slowly) through the majority of training, or at minimum the decay fraction should be tuned to match the actual expected training length.

## Suggested Fix

Either:
1. **Increase `epsilon_decay_fraction` to 0.6–0.8** — ensures decay continues through most of training
2. **Switch to step-based or episode-based decay** that continues as long as training continues
3. **Add a final epsilon floor** lower than 0.05 for late-training exploitation

```toml
epsilon_decay_fraction = 0.7   # decay through 70% of buffer capacity
```

## Labels

enhancement, hyperparameters

**Comment by ArgonZapan** [comment]
Won't fix: with batch_size=2048 and buffer_capacity=1000000, epsilon decays over 100 000 training steps = 204M experience draws. Sufficient coverage. Can be revisited via epsilon_decay_fraction in config if needed.
